oci/private/authn.bzl

```diff
@@ -201,16 +279,34 @@ def _get_token(rctx, state, registry, repository):
             # if a token for this repository and registry is acquired, use that instead.
             if url in state["token"]:
                 return state["token"][url]
+            
+            auth = None
+            if pattern["login"] == "<token>":
```

Comment:
Okay, this feels like the best we could do here. However, i don't want to commit this to the public api yet. Can we hide this feature behind a flag and print a fail statement that say something along the lines of;

```
OAuth2 support for oci_pull is highly experimental and is not enabled by default. 

We may change or abandon it without a notice. Use it at your own peril!

To enable this feature, add `common --repo_env=OCI_ENABLE_OAUTH2_SUPPORT` to the `.bazelrc` file.
```

Reply:
`rctx.getenv("OCI_ENABLE_OAUTH2_SUPPORT")` returns `None` for suggested `--repo_env=OCI_ENABLE_OAUTH2_SUPPORT`, so instead I suggest to add `common --repo_env=OCI_ENABLE_OAUTH2_SUPPORT=1`

Reply:
`rctx.getenv("OCI_ENABLE_OAUTH2_SUPPORT")` returns `None` for suggested `--repo_env=OCI_ENABLE_OAUTH2_SUPPORT`, so instead I suggest to add `common --repo_env=OCI_ENABLE_OAUTH2_SUPPORT=1`

Reply:
we probably don't want to use `rctx.getenv` as it a newer api and might not work on some 6.x versions of bazel. 

`rctx.os.environ` should do the trick, i am okay with the naming suggestion.